### DISCUSSION TITLE
Discrepancy between QGIS and Python OSGridConverter library

### DISCUSSION THREAD

**Discussion by halltony** [FieldStudiesCouncil/QGIS-Biological-Recording-Tools#69] [Q&A]
Hi,

I'm really enjoying using the Biological Recording Tools - they are a great plugins for analysing bird observation data.  Many thanks for developing and sharing this resource.

I am seeing a fairly sizeable discrepancy between the total number 1km squares identified by this plugin and some logic in a standalone python script using the OSGridConverter library.  Both approaches use the same lat and long fields from the same dataset.  Both are using WSG.

I wonder if the approach to identifying the squares might be different.  In python I am generating a full Grid Ref and then truncating that to a 1km square ref.  Does the plugin do something different/better?

I'll take a look at the code but though I'd ask in case anyone knows the answer off hand.

Thanks

**Comment by burkmarr** [comment]
I'd be surprised if there was an error of this kind in the tool - it's been around for the best part of 10 years with thousands of users and no systematic error of this kind has been reported. You never know though! Perhaps if you could try a sample of 20 or 30 six, eight or ten figure grid references - transforming to 1 km squares using the tool and your method and tabulate the results for comparison. 

A common mistake when converting grid references by truncating is to truncate the wrong positions. For example:
SD1234567890 should become SD1267, not SD1234.

**Reply by halltony** [reply]
Thanks for confirming the truncation logic. This is how I had performed it. As you suggest I will experiment with a few coordinates and compare outcomes.  There could be an issue with the Python library as well.
